docs/defradb/How-to Guides/peer-to-peer-how-to.md

```diff
@@ -0,0 +1,226 @@
+
+```bash
+defradb start --p2paddr /ip4/<ip_address>/tcp/<port>
+```
+
+Example:
+
+```bash
+defradb start --p2paddr /ip4/0.0.0.0/tcp/9172
+```
+
+**Parameters**:
+
+- Replace `<ip_address>` with your actual IP address (use `0.0.0.0` to listen on all interfaces)
+- Replace `<port>` with your desired port number
+
+## Manage Peer IDs
+
+### Get your Peer ID
+
+To retrieve your node's Peer ID using HTTP:
+
+```bash
+curl -H "Accept: application/json" http://localhost:9181/api/p2p/info
+```
+
+The Peer ID is generated from a private key created during the first startup and remains consistent across restarts.
+
+## Connect to peers
+
+### Connect to a specific peer
+
+To connect to a particular peer when starting DefraDB:
+
+```bash
+defradb start --peers /ip4/<ip_address>/tcp/<port>/p2p/<peer_id>
+```
+
+Example:
+
+```bash
+defradb start --peers /ip4/192.168.1.100/tcp/9171/p2p/12D3KooWEFCQ1iGMobsmNTPXb758kJkFc7XieQyGKpsuMxeDktz4
+```
+
+**Parameters**:
+
+- Replace `<ip_address>` with the peer's IP address
+- Replace `<port>` with the peer's P2P port
+- Replace `<peer_id>` with the peer's Peer ID
+
+## Manage document subscriptions (passive replication)
+
+Passive replication works at the document level. Subscribe to specific documents to receive updates automatically.
+
+### Subscribe to document updates
+
+```bash
+defradb client p2p document add <docID>
+```
+
+Example:
+
+```bash
+defradb client p2p document add bafybeihz5k3c2jzx7m4x5v6p7q8r9s0t1u2v3w4x5y6z7a8b9c0d1e2f3
+```
+
+### Unsubscribe from document updates
+
+```bash
+defradb client p2p document remove <docID>
+```
+
+### View all active document subscriptions
+
+```bash
+defradb client p2p document getall
+```
+
+## Manage collection subscriptions (active replication)
+
+Active replication can work at the collection level, allowing you to replicate entire collections to specific peers.
+
+### Subscribe to collection updates
+
+```bash
+defradb client p2p collection add <collectionID>
+```
+
+### Unsubscribe from collection updates
+
+```bash
+defradb client p2p collection remove <collectionID>
+```
+
+### View all active collection subscriptions
+
+```bash
+defradb client p2p collection getall
+```
+
+## Enable active replication
+
+Active replication creates a direct, persistent connection to a specific peer for reliable data synchronization.
+
+### Add a replicator using HTTP
+
+```bash
+curl -X POST http://localhost:9181/api/p2p/replicators \
+  -H "Content-Type: application/json" \
+  -d '{
+    "Info": {
+      "ID": "<peer_id>",
+      "Addrs": ["<peer_address>"]
+    },
+    "Collections": ["<collection_name>"]
+  }'
+```
+
+Example:
+
+```bash
+curl -X POST http://localhost:9181/api/p2p/replicators \
+  -H "Content-Type: application/json" \
+  -d '{
+    "Info": {
+      "ID": "12D3KooWEFCQ1iGMobsmNTPXb758kJkFc7XieQyGKpsuMxeDktz4",
+      "Addrs": ["/ip4/192.168.1.100/tcp/9171"]
+    },
+    "Collections": ["Books"]
+  }'
+```
```

Comment:
We can omit these `curl` methods for now. In the future, we can have every snippet have their curl vs CLI variation